src/dashboard/apps/consent/views.py

```diff
@@ -2,92 +2,102 @@
+        self._bulk_update_consent(selected_ids, VALIDATED)  # type: ignore
+        awaiting_ids = self._get_awaiting_ids(selected_ids)
+        self._bulk_update_consent(awaiting_ids, AWAITING)  # type: ignore
 
-    if request.POST:
-        selected_ids = request.POST.getlist("status")
-        update_consent_status(request.user, entities, selected_ids)
+        messages.success(self.request, _("Consents updated."))
 
-        messages.success(request, _("Consents updated."))
         return redirect(reverse("consent:index"))
 
-    breadcrumb_data = {
-        "links": [
-            {"url": reverse("consent:index"), "title": _("Consent")},
-        ],
-        "current": _("Manage Consents"),
-    }
+    def get_context_data(self, **kwargs):
+        """Add the user's entities to the context.
 
-    return render(
-        request=request,
-        template_name=template_name,
-        context={"entities": entities, "breadcrumb_data": breadcrumb_data},
-    )
+        Adds to the context the entities that the user has permission to access.
+        If a slug is provided, adds the entity corresponding to the slug.
+        """
+        context = super().get_context_data(**kwargs)
+        context["entities"] = self._get_entities()
+        return context
 
+    def _get_entities(self) -> list:
+        """Return a list of entities or specific entity if slug is provided."""
+        slug: str | None = self.kwargs.get("slug", None)
+        user: DashboardUser = self.request.user  # type: ignore
 
-def update_consent_status(user, entities, selected_ids):
-    """Updates the status of consents.."""
+        if slug:
+            entity: Entity = get_object_or_404(Entity, slug=slug)
+            if not user.can_validate_entity(entity):
+                raise PermissionDenied
+            return [entity]
+        else:
+            return list(user.get_entities())
 
-    def _bulk_update_consent(ids: list[str], status: str):
+    def _bulk_update_consent(self, ids: list[str], status: str) -> int:
         """Bulk update of the consent status for a given status and list of entities."""
-        Consent.objects.filter(id__in=ids).update(
-            status=status,
-            created_by=user,
-            updated_at=timezone.now(),
+        return (
+            Consent.objects.filter(id__in=ids)
+            .filter(
+                Q(delivery_point__entity__users=self.request.user)
+                | Q(delivery_point__entity__proxies__users=self.request.user)
+            )
+            .update(
+                status=status,
+                created_by=self.request.user,
+                updated_at=timezone.now(),
+            )
         )
 
-    def _get_awaiting_ids(entities, ids):
-        """Get the a list of the non selected ids (awaiting ids)."""
-        base_ids = []
+    def _get_awaiting_ids(self, validated_ids: list[str]) -> list[str]:
+        """Get the list of the non-selected IDs (awaiting IDs)."""
```

Comment:
Friday one-liner :boom: 

```suggestion
    def _get_awaiting_ids(self, validated_ids: list[str]) -> list[str]:
        """Get the list of the non-selected IDs (awaiting IDs)."""
        
        return [
            str(c.id) for e in self.get_entities() for c in e if str(c.id) not in validated_ids
        ]
```